test/server/connection_handler_test.cc

```diff
@@ -379,7 +379,18 @@ TEST_F(ConnectionHandlerTest, NormalRedirect) {
   EXPECT_CALL(dispatcher_, createServerConnection_(_, _)).WillOnce(Return(connection));
   EXPECT_CALL(factory_, createNetworkFilterChain(_, _)).WillOnce(Return(true));
   listener_callbacks1->onAccept(Network::ConnectionSocketPtr{accepted_socket}, true);
+
+  // Verify per-listener connection stats.
   EXPECT_EQ(1UL, handler_->numConnections());
+  EXPECT_EQ(1UL, TestUtility::findCounter(stats_store_, "downstream_cx_total")->value());
+  EXPECT_EQ(1UL, TestUtility::findGauge(stats_store_, "downstream_cx_active")->value());
+  EXPECT_EQ(1UL, TestUtility::findCounter(stats_store_, "test.downstream_cx_total")->value());
```

Comment:
Do we have tests that runs with more than 1 worker threads and checks both per-worker and aggregate stats?

Reply:
We don't have any tests in the code base the run with more than 1 worker unfortunately. I think this is a worthwhile thing to consider but do you mind if I open up a tech debt follow up issue?

Reply:
A TODO might be good here for that -- or file an issue. Either way.

Reply:
TODO sgtm